tests/unit/User/ManagerTest.php

```diff
@@ -550,4 +550,49 @@ public function testResolveUID() {
 
 		self::assertSame('lastname', $this->manager->resolveUID($entry));
 	}
+
+	public function testDnUpdateWhenUuidMatches() {
+		$oldDn = 'cn=olddn,ou=users,dc=example,dc=com';
+		$newDn = 'cn=newdn,ou=users,dc=example,dc=com';
+		$uuid = 'aaaa-bbbb-cccc-dddd';
+		$mapper = $this->createMock(UserMapping::class);
+		$mapper->expects($this->once())
+			->method('getUUIDByDN')
+			->with($oldDn)
+			->willReturn($uuid);
+
+		$this->access->expects($this->any())
+			->method('getUserMapper')
+			->willReturn($mapper);
+		$this->access->expects($this->any())
+			->method('getUserDnByUuid')
+			->with($uuid)
+			->willReturn($newDn);
+		$this->access->expects($this->any())
+			->method('readAttribute')
+			->willReturn([]);
+		$isRemapped = $this->manager->dnExistsOnLDAP($oldDn);
+		$this->assertTrue($isRemapped);
+	}
+
+	public function testDnNotUpdatedWhenUuidNotFound() {
+		$oldDn = 'cn=olddn,ou=users,dc=example,dc=com';
+		$newDn = 'cn=newdn,ou=users,dc=example,dc=com';
+		$uuid = 'aaaa-bbbb-cccc-dddd';
+		$mapper = $this->createMock(UserMapping::class);
+		$mapper->expects($this->once())
+			->method('getUUIDByDN')
+			->with($oldDn)
+			->willReturn($uuid);
+
+		$this->access->expects($this->any())
+			->method('getUserMapper')
+			->willReturn($mapper);
+		$this->access->expects($this->any())
+			->method('getUserDnByUuid')
+			->with($uuid)
+			->willReturn($newDn);
```

Comment:
Maybe we should be a bit more explicit, specially because this is a copy and paste of the test above for the most part (which is totally fine)
```
$this->access->expects($this->any())
			->method('readAttribute')
			->willReturn(null);
```

Reply:
done